lib/akamai_rspec/matchers/x_cache_headers.rb

```diff
@@ -16,10 +16,14 @@ def cache_headers
 
 RSpec::Matchers.define :be_served_from_origin do |contents|
   include AkamaiRSpec::CacheHelpers
+
   match do |url|
-    response = AkamaiRSpec::Request.get url
-    response.headers.any? { |key, value| x_cache_headers.include?(key) && value =~ /\/#{contents}\// } && \
-      response.code == 200
+    @response = AkamaiRSpec::Request.get_with_debug_headers url
+    @response.code == 200 && cache_headers.any? {|h| h.split("/").include? contents}
```

Comment:
Instead of broadly looking for the origin name in any header can we not just look for it in the specific Akamai header that specifies the origin (and fail if this header is missing)?

Otherwise if Akamai insert the origin into some other header for an unrelated reason this could produce a false positive.